Gems/Multiplayer/Code/CMakeLists.txt

```diff
@@ -35,6 +34,89 @@ ly_add_target(
         *.AutoPackets.xml,AutoPackets_Source.jinja,$path/$fileprefix.AutoPackets.cpp
         *.AutoPackets.xml,AutoPacketDispatcher_Header.jinja,$path/$fileprefix.AutoPacketDispatcher.h
         *.AutoPackets.xml,AutoPacketDispatcher_Inline.jinja,$path/$fileprefix.AutoPacketDispatcher.inl
+)
+
+ly_add_target(
+    NAME Multiplayer.Client.Static STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        multiplayer_files.cmake
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            ${pal_source_dir}
+            Source
+            .
+        PUBLIC
+            Include
+    COMPILE_DEFINITIONS
+        PUBLIC
+            AZ_TRAIT_CLIENT_ENABLED=1
+            AZ_TRAIT_SERVER_ENABLED=0
+    BUILD_DEPENDENCIES
+        PUBLIC
+            Gem::Multiplayer.AutoGen.Static
+        PRIVATE
+            Gem::EMotionFXStaticLib
+            Gem::PhysX.Static
+    AUTOGEN_RULES
+        *.AutoComponent.xml,AutoComponent_Header.jinja,$path/$fileprefix.AutoComponent.h
```

Comment:
I see you have the autogen target above now, doesn't that mean you can remove the autogen regex expressions from the other targets?

Reply:
No they're actually doing different thing. AutoGen is providing all the non-split codegen work (Packets mostly). The other targets are doing Components. This is so the set of split logic we're generating per platform is smaller.

Reply:
I've opted to remove the AutoGen target and just have each target own everything. This leaves the door open to split or other macro based logic in those templates and cleans the CMake up a bit..